workflow/executor/agent.go

```diff
@@ -298,11 +310,12 @@ func (ae *AgentExecutor) executeHTTPTemplateRequest(ctx context.Context, httpTem
 		}
 		request.Header.Add(header.Name, value)
 	}
-	httpClient := http.DefaultClient
+
 	if httpTemplate.TimeoutSeconds != nil {
-		httpClient.Timeout = time.Duration(*httpTemplate.TimeoutSeconds) * time.Second
+		httpClients[httpTemplate.InsecureSkipVerify].Timeout = time.Duration(*httpTemplate.TimeoutSeconds) * time.Second
```

Comment:
This is a bug, we should never modify the HTTP client like this. We should fix this.

Please research "golang http timeout", I think you need to use NewRequestWithTimeout?

Reply:
Will look into this.

Reply:
A solution I've found is changing the timeout per request. 

```
if httpTemplate.TimeoutSeconds != nil {
		ctx, cancel := context.WithTimeout(ctx, time.Duration(*httpTemplate.TimeoutSeconds)*time.Second)
		defer cancel()
		request = request.WithContext(ctx)
	} else {
		request = request.WithContext(ctx)
	}
```